src/services/goToDefinition.ts

```diff
@@ -28,36 +28,69 @@ namespace ts.GoToDefinition {
 
         if (isStaticModifier(node) && isClassStaticBlockDeclaration(node.parent)) {
             const classDecl = node.parent.parent;
-            const symbol = getSymbol(classDecl, typeChecker);
+            const { symbol, failedAliasResolution } = getSymbol(classDecl, typeChecker);
+
             const staticBlocks = filter(classDecl.members, isClassStaticBlockDeclaration);
             const containerName = symbol ? typeChecker.symbolToString(symbol, classDecl) : "";
             const sourceFile = node.getSourceFile();
             return map(staticBlocks, staticBlock => {
                 let { pos } = moveRangePastModifiers(staticBlock);
                 pos = skipTrivia(sourceFile.text, pos);
-                return createDefinitionInfoFromName(typeChecker, staticBlock, ScriptElementKind.constructorImplementationElement, "static {}", containerName, { start: pos, length: "static".length });
+                return createDefinitionInfoFromName(typeChecker, staticBlock, ScriptElementKind.constructorImplementationElement, "static {}", containerName, /*unverified*/ false, failedAliasResolution, { start: pos, length: "static".length });
             });
         }
 
-        const symbol = getSymbol(node, typeChecker);
+        let { symbol, failedAliasResolution } = getSymbol(node, typeChecker);
+        let fallbackNode = node;
+
+        if (searchOtherFilesOnly && failedAliasResolution) {
+            // We couldn't resolve the specific import, try on the module specifier.
+            const importDeclaration = findAncestor(node, isAnyImportOrBareOrAccessedRequire);
+            const moduleSpecifier = importDeclaration && tryGetModuleSpecifierFromDeclaration(importDeclaration);
+            if (moduleSpecifier) {
+                ({ symbol, failedAliasResolution } = getSymbol(moduleSpecifier, typeChecker));

```

Comment:
Nit: what's up with the parens?

Reply:
The binding pattern parses as a block without them. This construct needs to be forced into an expression construct.